pkg/gadget-collection/gadgets/traceloop/gadget.go

```diff
@@ -52,138 +67,298 @@ some differences:
 
 * traceloop uses eBPF instead of ptrace
 * traceloop's tracing granularity is the container instead of a process
-* traceloop's traces are recorded in a fast, in-memory, overwritable ring
+* traceloop's traces are recorded in a fast, in-memory, over writable ring
   buffer like a flight recorder. The tracing could be permanently enabled and
   inspected in case of crash.
 `
 }
 
 func (f *TraceFactory) OutputModesSupported() map[gadgetv1alpha1.TraceOutputMode]struct{} {
 	return map[gadgetv1alpha1.TraceOutputMode]struct{}{
-		gadgetv1alpha1.TraceOutputModeExternalResource: {},
+		gadgetv1alpha1.TraceOutputModeStatus: {},
+		gadgetv1alpha1.TraceOutputModeStream: {},
 	}
 }
 
 func deleteTrace(name string, t interface{}) {
 	trace := t.(*Trace)
-	trace.stop()
+	if trace.started {
+		trace.helpers.Unsubscribe(genPubSubKey(name))
```

Comment:
`deleteTrace()` needs to stop the tracer either by calling `Stop()` or by calling `traceUnique.tracer.Stop()` directly.

This is because there is 2 different code paths where the traceloop gadget is stopped:
1. The CLI calls the "stop" operation and then deletes the Trace CR (via `kubectl-gadget traceloop stop`).
2. The CLI deletes the Trace CR directly (via `kubectl delete trace -n gadget traceloop-7j5bj`).

See how `pkg/gadget-collection/gadgets/advise/seccomp/gadget.go` is calling `traceSingleton.tracer.Close()` at two different places.

This problem leads to temporary file descriptor leaks (until the garbage collector closes them). Some gadgets have a unit test for this (see `checkFdList()` in `pkg/local-gadget-manager/local-gadget-manager_test.go`). But traceloop does not seem to have a test for this.

Tasks:
- [ ] Fix the fd leak for the 2nd code path.
- [ ] Add a unit test for the fd leak like other gadgets

This can be done in the next PR.